File: CoreAngularDemo.Client/src/app/modules/shared/models/work-type.ts
import {TEntity} from '../../core/models/entity/entity';

export class WorkType extends TEntity<WorkType> {
    id: number;
    name: string;
    estimatedCost: number;
    estimatedTime: number;

    constructor(workType: Partial<WorkType>) {
        super(workType);
    }
}

File: CoreAngularDemo.API/CoreAngularDemo.DAL/Models/Entities/WorkType.cs
using System;
using System.Collections.Generic;
using CoreAngularDemo.DAL.Models.Entities.Abstractions;

namespace CoreAngularDemo.DAL.Models.Entities
{
    public class WorkType : IAuditableEntity, IBaseEntity
    {
        public WorkType()
        {
            IssueLog = new HashSet<IssueLog>();
        }

        public int Id { get; set; }
        public string Name { get; set; }
        public double EstimatedTime { get; set; }
        public double EstimatedCost { get; set; }
        public DateTime? CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }
        public string CreatedById { get; set; }
        public string UpdatedById { get; set; }

        public virtual User Create { get; set; }
        public virtual User Mod { get; set; }

        public virtual ICollection<IssueLog> IssueLog { get; set; }
    }
}
